File: snapshot/2009/src/rac/labs/vb/WaitCursor.vb
Imports System.Windows.Forms

Namespace Labs
    Class WaitCursor
        Dim _oldCursor As Cursor

        Public Sub New()
            _oldCursor = Cursor.Current
            Cursor.Current = Cursors.WaitCursor
        End Sub

        Protected Overrides Sub Finalize()
            Cursor.Current = _oldCursor
        End Sub
    End Class
End Namespace

File: snapshot/2014/adn/src/rme/AdnRme/AdnRme/WaitCursor.cs
using System.Windows.Forms;

namespace AdnRme
{
  class WaitCursor
  {
    Cursor _oldCursor;

    public WaitCursor()
    {
      _oldCursor = Cursor.Current;
      Cursor.Current = Cursors.WaitCursor;
    }

    ~WaitCursor()
    {
      Cursor.Current = _oldCursor;
    }
  }
}
